src/compiler/utilities.ts

```diff
@@ -3504,6 +3504,40 @@ namespace ts {
     export function getCombinedLocalAndExportSymbolFlags(symbol: Symbol): SymbolFlags {
         return symbol.exportSymbol ? symbol.exportSymbol.flags | symbol.flags : symbol.flags;
     }
+
+    export function isWriteOnlyAccess(node: Node) {
+        return isWriteAccessWorker(node, /*writeOnly*/ true);
+    }
+
+    export function isReadOnlyAccess(node: Node) {
+        return !isWriteAccessWorker(node, /*writeOnly*/ false);
+    }
+
+    /**
+     * @param writeOnly If set, return true for `x++;` but not for `f(x++);`, which uses `x` in addition to writing to it.
+     */
+    function isWriteAccessWorker(node: Node, writeOnly: boolean): boolean {

```

Comment:
`writeOnly` feels like a misnomer, since, based on its usage with `isWriteOnlyAccess` and `isReadOnlyAccess`, the parameter toggles between `readonly` and `writeonly` functionality. Maybe the function should be called `accessCheckWorker` and `isWriteAccess`?

Reply:
`isReadOnlyAccess` returns `!isWriteAccessWorker(node, /*writeOnly*/ false);` -- `isWriteAccessWorker(node, false);`  is true for *any* write, so its negation is whether the access is read-only.
I've changed this to an enum-returning function so there are fewer negations and it's easier to read.